site/content/en/latest/design/eg-metrics.md

```diff
@@ -0,0 +1,634 @@
+---
+
+This document aims to cover all aspects of envoy gateway control plane metrics observability.
+
+{{% alert title="Note" color="secondary" %}}
+**Data plane** observability (while important) is outside of scope for this document.
+{{% /alert %}}
+
+## Current State
+
+At present, the Envoy Gateway control plane provides logs and controller-runtime metrics, without traces. Logs are managed through our proprietary library (`internal/logging`, a shim to `zap`) and are written to `/dev/stdout`.
+
+The absence of comprehensive and robust control plane metrics observability hinders the effective monitoring of Envoy Gateway in a production environment, a critical requirement before deploying Envoy Gateway into production.
+
+## Goals
+
+Our objectives include:
+
++ Supporting **PULL** mode for Prometheus metrics and exposing these metrics on the admin address.
++ Supporting **PUSH** mode for Prometheus metrics, thereby sending metrics to the Open Telemetry Stats sink.
++ Offering a COMMON metrics library so developers can effortlessly add new metrics/labels to each Envoy Gateway component.
++ Providing BASIC metrics produced by each Envoy Gateway component, including:
+  + Provider
+  + Resource Translator
+  + Infra Manager
+  + xDS Translator
+  + Extension Manager
+
+## Non-Goals
+
+Our non-goals include:
+
++ Supporting other stats sinks.
+
+## Use-Cases
+
+The use-cases include:
+
++ Exposing Prometheus metrics in the Envoy Gateway Control Plane.
++ Pushing Envoy Gateway Control Plane metrics via the Open Telemetry Sink.
+
+## Design
+
+### Standards
+
+Our metrics, and traces in the future, will be built upon the [OpenTelemetry](https://opentelemetry.io/) standards. All metrics will be configured via the [OpenTelemetry SDK](https://opentelemetry.io/docs/specs/otel/metrics/sdk/), which offers neutral libraries that can be connected to various backends.
+
+This approach allows the Envoy Gateway code to concentrate on the crucial aspect - generating the metrics - and delegate all other tasks to systems designed for telemetry ingestion.
+
+### Attributes
+
+OpenTelemetry defines a set of [Semantic Conventions](https://opentelemetry.io/docs/concepts/semantic-conventions/), including [Kubernetes specific ones](https://opentelemetry.io/docs/specs/otel/resource/semantic_conventions/k8s/).
+
+These attributes can be expressed in logs (as keys of structured logs), traces (as attributes), and metrics (as labels).
+
+We aim to use attributes consistently where applicable. Where possible, these should adhere to codified Semantic Conventions; when not possible, they should maintain consistency across the project.
+
+### Extensibility
+
+Envoy Gateway supports both **PULL/PUSH** mode metrics, with Metrics exported via Prometheus by default.
+
+Additionally, Envoy Gateway can export metrics using the [OTEL HTTP metrics exporter](https://opentelemetry.io/docs/specs/otel/metrics/sdk_exporters/otlp/#general), which pushes metrics to a remote OTEL collector.
```

Comment:
should be start off with GRPC exporter since that is what we use by default in data plane
https://github.com/envoyproxy/gateway/blob/57b00ed7e01d69b479307f3996fd7878cb985fbf/internal/xds/bootstrap/bootstrap.yaml.tpl#L112

Reply:
Sure, I have added that.